rest/src/test/kotlin/systems/ajax/englishstudytelegrambot/NatsControllerTest.kt

```diff
@@ -0,0 +1,217 @@
+package systems.ajax.englishstudytelegrambot
+
+import io.nats.client.Connection
+import io.nats.client.Message
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.remove
+import response_request.CreateNewLibrary.CreateNewLibraryRequest
+import response_request.CreateNewLibrary.CreateNewLibraryResponse
+import response_request.DeleteLibrary
+import response_request.DeleteLibrary.DeleteLibraryRequest
+import response_request.DeleteLibrary.DeleteLibraryResponse
+import response_request.GetAllLibraries.GetAllLibrariesRequest
+import response_request.GetAllLibraries.GetAllLibrariesResponse
+import response_request.GetAllUsers.GetAllUsersRequest
+import response_request.GetAllUsers.GetAllUsersResponse
+import response_request.GetAllWords.GetAllWordsRequest
+import response_request.GetAllWords.GetAllWordsResponse
+import response_request.GetAllWordsFromLibrary.GetAllWordsFromLibraryRequest
+import response_request.GetAllWordsFromLibrary.GetAllWordsFromLibraryResponse
+import systems.ajax.NatsSubject.CREATE_NEW_LIBRARY
+import systems.ajax.NatsSubject.DELETE_LIBRARY
+import systems.ajax.NatsSubject.GET_ALL_LIBRARIES_SUBJECT
+import systems.ajax.englishstudytelegrambot.entity.Library
+import systems.ajax.englishstudytelegrambot.entity.User
+import systems.ajax.englishstudytelegrambot.entity.Word
+import systems.ajax.englishstudytelegrambot.service.LibraryService
+import systems.ajax.NatsSubject.GET_ALL_USERS_SUBJECT
+import systems.ajax.NatsSubject.GET_ALL_WORDS_FROM_LIBRARY
+import systems.ajax.NatsSubject.GET_ALL_WORDS_SUBJECT
+import systems.ajax.englishstudytelegrambot.controller.AdminController
+import systems.ajax.englishstudytelegrambot.dto.WordDto
+import systems.ajax.englishstudytelegrambot.service.AdminService
+import systems.ajax.englishstudytelegrambot.service.WordService
+import java.time.Duration
+
+@SpringBootTest
+class NatsControllerTest {
+
+    @Autowired
+    private lateinit var natsConnection: Connection
+
+    @Autowired
+    private lateinit var libraryService: LibraryService
```

Comment:
Is there a reason to autowire **Service** if you add predefined @Validated data?  I think you should use **Repository** instead.